solutuins-1/src/main/java/solutions_1/solutuins_1/NonReducible.java

```diff
@@ -0,0 +1,103 @@
+package solutions_1.solutuins_1;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The Class NonReducible.
+ */
+
+public class NonReducible {
+
+	/** The logger. */
+	private Logger logger;
+
+	/**
+	 * Instantiates a new NonReducible
+	 */
+	public NonReducible() {
+		logger = Logger.getLogger(NonReducible.class.getSimpleName());
+	}
+
+	/**
+	 * Trims the input string as much as possible.
+	 *
+	 * @param String the input string
+	 * @return The object ChopCount containing the trimmed string and number of operations
+	 */
+	public ChopCount getChopCount(String input) {
+		ChopCount chopCount = new ChopCount();
+		try {
+			int totalChops = 0;
+			for(int i=0;i<input.length();i++) {
+				if(i==0) {
+					continue;
+				}
+				if(input.charAt(i) == input.charAt(i-1)) {
+					input = input.substring(0, i-1) + input.substring(i+1,input.length());
+					i--;
+					totalChops++;
+				}
+			}
+			chopCount.setNonReducableStr(input);
+			chopCount.setNumberOfChops(totalChops);
+		} catch(Exception e) {
+			logger.log(Level.SEVERE, " NonReducible::getChopCount()");
+		}
+		return chopCount;
+	}
+}
+
+/**
+ * The Class ChopCount.
+ */
+class ChopCount {

```

Comment:
why we need a class here? Can we achieve this without this class?

Reply:
In the problem statement, the expected output was an object with reduced string and number of operation so I had created such class. We can just return an anonymous class Object like /* Object obj = new Object(){ //reduced string and number of operations variables}; */

Reply:
Can you please explain this anonymous class object.
Not able to comprehend your explanation.

Reply:
![anonymous_obj](https://user-images.githubusercontent.com/3712951/41608370-02a7466a-7406-11e8-80b4-f191cfd76c81.png)
Hi Neeraj,

Please find the attached image for the object that we can return instead of creating a ChopCount class.

Regards,
Arjun

Reply:
Thanks

Reply:
One more way is to just have those variables in NonReducible class itself and return the same instead of having a separate ChopCount class.